Libraries/LibWeb/IndexedDB/Internal/Algorithms.cpp

```diff
@@ -1869,4 +1870,32 @@ WebIDL::ExceptionOr<JS::Value> retrieve_a_key_from_an_object_store(JS::Realm& re
     return convert_a_key_to_a_value(realm, record.value().key);
 }
 
+// https://w3c.github.io/IndexedDB/#retrieve-multiple-values-from-an-object-store
+WebIDL::ExceptionOr<JS::Value> retrieve_multiple_values_from_an_object_store(JS::Realm& realm, GC::Ref<ObjectStore> store, GC::Ref<IDBKeyRange> range, Optional<WebIDL::UnsignedLong> count_parameter)
+{
+    // 1. If count is not given or is 0 (zero), let count be infinity.
+    size_t count = count_parameter.has_value() ? count_parameter.value() : AK::NumericLimits<WebIDL::UnsignedLong>::max();
+
+    // 2. Let records be a list containing the first count records in store’s list of records whose key is in range.
+    auto records = store->first_n_in_range(range, count);
+
+    // 3. Let list be an empty list.
+    Vector<JS::Value> list;
```

Comment:
It is not safe to be storing JS::Value's on the heap like this as there could be a GC that happens inside the foor loop below, and nothing protects this list from GC. Probably simplest to just make this a JS::Array up front